CK3toEU4/Source/Mappers/DynamicIdeasMapper/DynamicIdeasLink.cpp

```diff
@@ -0,0 +1,53 @@
+#include "DynamicIdeasLink.h"
+#include "CommonRegexes.h"
+#include "Log.h"
+#include "ParserHelpers.h"
+
+mappers::DynamicIdeasLink::DynamicIdeasLink(std::istream& theStream)
+{
+	registerKeys();
+	parseStream(theStream);
+	clearRegisteredKeywords();
+}
+
+void mappers::DynamicIdeasLink::registerKeys()
+{
+	registerKeyword("ethos", [this](std::istream& theStream) {
+		ethos = commonItems::getString(theStream);
+	});
+	registerKeyword("tradition", [this](std::istream& theStream) {
+		tradition = commonItems::getString(theStream);
+	});
+	registerKeyword("default", [this](std::istream& theStream) {
+		defaultString = commonItems::getString(theStream);
+	});
+	registerKeyword("idea_name", [this](std::istream& theStream) {
+		ideaName = commonItems::getString(theStream);
+	});
+	registerKeyword("ck3", [this](std::istream& theStream) {
+		const auto scraper = DynamicIdeasLink(theStream);
+		if (auto possibleEthos = scraper.getEthos(); possibleEthos)
+			ethos = possibleEthos;
+		else if (auto possibleTradition = scraper.getTradition(); possibleTradition)
+			tradition = possibleTradition;
+		else if (auto possibleDefault = scraper.getDefault(); possibleDefault)
+			defaultString = possibleDefault;
+		else
+			Log(LogLevel::Info) << "Missing value in tradition_ideas.txt";
+	});
+	registerKeyword("eu4", [this](std::istream& theStream) {
+		const auto scraper = DynamicIdeasLink(theStream);
```

Comment:
At the moment no, because there is a case where order of the assignment matters, although I'm considering changing that. it would require a minor break from vanilla convention.

Reply:
Because national ambitions always have a single effect (with the sole exception of Hawaiian?) I truncate ideas that would have 2 effects if they end up in the ambition slot. Order matters so the truncation happens in a reasonable way. (Most beneficial effect stays).

Reply:
If I just stopped worrying about it, then yes I could use commonItems::assignments and remove the recursive bits/ extra regex
```
registerKeyword("eu4", [this](std::istream& theStream) {
	const auto scraper = commonItems::assignments(theStream).getAssignments();
	for (const auto& [type, value]: scraper)
		effects.push_back({type, value}); // Convert std::pair to Assignment Pair
});
registerKeyword("rule", [this](std::istream& theStream) {
	const auto scraper = commonItems::assignments(theStream).getAssignments();
	for (const auto& [type, value]: scraper)
	{
		if (type == "idea_name")
			ideaName = value;
		else
			rules.push_back({type, value}); // Convert std::pair to Assignment Pair
	}
});
```